### ISSUE TITLE
Brightness problem in SRGAN

### ISSUE THREAD

**Issue by YUFEI96HE** [eriklindernoren/PyTorch-GAN#54]
I notice that the generated images have higher brightness and more colors than the original image, or the resulting images of other approaches. What causes it?

**Comment by chenyang1999** [comment]
maybe you can try this code:
`
# Normalization parameters for pre-trained PyTorch models
mean = np.array([0.485, 0.456, 0.406])
std = np.array([0.229, 0.224, 0.225])


def denormalize(tensors):
    """ Denormalizes image tensors using mean and std """
    for c in range(3):
        tensors[:, c].mul_(std[c]).add_(mean[c])
    return torch.clamp(tensors, 0, 255)
`
`
if batches_done % opt.sample_interval == 0:
            # Save image grid with upsampled inputs and SRGAN outputs
            imgs_lr = nn.functional.interpolate(imgs_lr, scale_factor=4)
            img_grid = denormalize(torch.cat((imgs_lr, gen_hr), -1))
            save_image(img_grid, "images/training/%d.png" % batches_done, nrow=1, normalize=False)
`
in srgan.py

`
class ImageDataset(Dataset):
    def __init__(self, root, hr_shape):
        hr_height, hr_width = hr_shape
        # Transforms for low resolution images and high resolution images
        self.lr_transform = transforms.Compose(
            [
                transforms.Resize((hr_height // 4, hr_height // 4), Image.BICUBIC),
                transforms.ToTensor(),
                transforms.Normalize(mean, std),
            ]
        )
        self.hr_transform = transforms.Compose(
            [
                transforms.Resize((hr_height, hr_height), Image.BICUBIC),
                transforms.ToTensor(),
                transforms.Normalize(mean, std),
            ]
        )

        self.files = sorted(glob.glob(root + "/*.*"))

    def __getitem__(self, index):
        img = Image.open(self.files[index % len(self.files)])
        img_lr = self.lr_transform(img)
        img_hr = self.hr_transform(img)

        return {"lr": img_lr, "hr": img_hr}

    def __len__(self):
        return len(self.files)
`
in datasets.py

**Comment by eriklindernoren** [comment]
@YUFEI96HE Hm, not sure. Would have to investigate this. Have you tried https://github.com/eriklindernoren/PyTorch-GAN/blob/master/implementations/esrgan/esrgan.py? Should produce better results.

**Comment by usmanovaanastasia** [comment]
Good day! I have the same problem, was it managed somehow?

**Comment by lizhuoq** [comment]
> Good day! I have the same problem, was it managed somehow?


I speculate that the ratio between pixel-wise loss and perceptual loss is too small. You should increase the weight of the pixel-wise loss or decrease the weight of the perceptual loss. I've also encountered this situation. If you have a solution, please let me know. Thank you!

**Comment by lizhuoq** [comment]
Alternatively, there is another approach. I speculate that increasing the warmup_batches to ten times its original value might alleviate this issue.
